core/storage/blog/gae_models.py

```diff
@@ -334,15 +334,62 @@ def has_reference_to_user_id(cls, user_id):
             cls.editor_ids == user_id).get(keys_only=True) is not None
 
     @classmethod
-    def get_by_user(cls, user_id):
-        """Retrieves the rights object for all blog posts assigned to given user
+    def get_multi_published_models_by_user(cls, user_id, limit=None):
+        """Retrieves the blog post rights objects for published blog posts for
+        which the given user is an editor .
 
         Args:
-            user_id: str. ID of user.
+            user_id: str. ID of the author of the blog post.
+            limit: int|None. Number of BlogPostRightsModel to be fetched.If
+                None, all existing published models by user will be fetched.
 
         Returns:
             list(BlogPostRightsModel). The list of BlogPostRightsModel objects
-            in which the given user is a editor.
+            in which the given user is an editor.
+        """
+        if limit is not None:
+            return cls.query(
+                cls.editor_ids == user_id,
+                cls.blog_post_is_published == True # pylint: disable=singleton-comparison
+            ).order(-cls.last_updated).fetch(limit)
+        return cls.query(
+            cls.editor_ids == user_id,
+            cls.blog_post_is_published == True).fetch() # pylint: disable=singleton-comparison
```

Comment:
Wouldn't you also order by cls.last_updated here?

If you do, then you can compute the query object (before calling fetch() on it) and then reuse it, thus avoiding duplicate code.

Also, if you are ordering by last updated, mention that in the docstring (ditto elsewhere) since it's part of the definition of the return value.

Reply:
Done.